addons/im_livechat/models/im_livechat_channel_member_history.py

```diff
@@ -90,3 +153,81 @@ def _compute_display_name(self):
     def _compute_avatar_128(self):
         for history in self:
             history.avatar_128 = history.partner_id.avatar_128 or history.guest_id.avatar_128
+
+    # ===================================================================
+    # REPORTING
+    # ===================================================================
+
+    @api.depends("call_history_ids")
+    def _compute_has_call(self):
+        for history in self:
+            history.has_call = 1 if history.call_history_ids else 0
+
+    @api.depends("call_history_ids.duration_hour")
+    def _compute_call_duration_hour(self):
+        for history in self:
+            history.call_duration_hour = sum(history.call_history_ids.mapped("duration_hour"))
+
+    @api.depends(
+        "channel_id.livechat_agent_requesting_help_history",
+        "channel_id.livechat_agent_providing_help_history",
+    )
+    def _compute_help_status(self):
+        agent_histories = self.filtered(lambda h: h.livechat_member_type == "agent")
+        (self - agent_histories).help_status = "none"
+        for history in agent_histories:
+            if (
+                history.channel_id.livechat_agent_requesting_help_history.partner_id
+                == history.partner_id
+            ):
+                history.help_status = "requested"
+            elif (
+                history.channel_id.livechat_agent_providing_help_history.partner_id
+                == history.partner_id
+            ):
+                history.help_status = "provided"
+
+    @api.depends("channel_id.rating_ids")
+    def _compute_rating_id(self):
+        agent_histories = self.filtered(lambda h: h.livechat_member_type == "agent")
+        (self - agent_histories).rating_id = None
```

Comment:
Wouldn't it possibly make sense to also take the rating for bots?
As some of them might lead to better outcome than others.